packages/back-end/src/services/experiments.ts

```diff
@@ -823,6 +826,115 @@ export function toSnapshotApiInterface(
   };
 }
 
+/**
+ * Get a Partial<MetricInterface> from a Partial<ApiMetric>
+ * This method does not perform any validation; it is expected that valid data be passed in,
+ * e.g. specifying only one of mixpanel, sql, sql.builder
+ * @param organization
+ * @param datasource
+ * @param apiMetric
+ * @return metric Partial<MetricInterface>
+ */
+export type RequiredApiMetricFields = Pick<
+  ApiMetric,
+  "datasourceId" | "type" | "name"
+>;
+export function partialFromMetricApiInterface(
+  organization: OrganizationInterface,
+  apiMetric: Partial<ApiMetric> & RequiredApiMetricFields,
+  dataSource: DataSourceInterface
+): Partial<MetricInterface> {
+  const metricDefaults = organization.settings?.metricDefaults;
+
+  const {
+    datasourceId,
+    name,
+    description,
+    type,
+    behavior,
+    sql,
+    mixpanel,
+    owner,
+    tags = [],
+    projects = [],
+  } = apiMetric;
+
+  let queryFormat: undefined | "builder" | "sql" = undefined;
+
+  const sqlBuilder = sql?.builder;
+  if (sqlBuilder) {
+    queryFormat = "builder";
+  } else if (sql) {
+    queryFormat = "sql";
+  }
+
+  const conditionsForDataSourceType = (): Condition[] => {
+    switch (dataSource.type) {
+      case "mixpanel":
+        return (mixpanel?.conditions || []).map(
+          ({ operator, property, value }) => ({
+            column: property,
+            operator: operator as Operator,
+            value: value,
+          })
+        );
+
+      default:
+        return (sqlBuilder?.conditions || []) as Condition[];
+    }
+  };
+
+  const metric: Omit<MetricInterface, "dateCreated" | "dateUpdated" | "id"> = {
+    datasource: datasourceId,
+    type: type as MetricType,
+    organization: organization.id,
+    name,
+    queryFormat,
+    description: description || "",
+    runStarted: null,
+    owner: owner || "",
+    tags,
+    projects,
+    inverse: behavior?.goal === "decrease",
+    ignoreNulls: false,
+    aggregation: sql?.userAggregationSQL || mixpanel?.userAggregation || "",
+    sql: sql?.userAggregationSQL || sql?.conversionSQL, // TODO: Is this correct?
+    cap: behavior?.cap ?? 0,
+    conditions: conditionsForDataSourceType(),
+
+    conversionDelayHours: behavior?.conversionWindowStart ?? 0,
+    conversionWindowHours: behavior?.conversionWindowEnd ?? 0,
```

Comment:
This needs to be `conversionWindowHours:  behavior?.conversionWindowEnd - behavior?.conversionWindowStart`

Reply:
Since both of those fields are optional, what do you think about one of these options:

With default conversion window (72):

```ts
    conversionWindowHours:
      (behavior?.conversionWindowEnd || DEFAULT_CONVERSION_WINDOW_HOURS) -
      (behavior?.conversionWindowStart ?? 0),
```

Or with 0 fallback:

```ts
    conversionWindowHours:
      (behavior?.conversionWindowEnd || 0) -
      (behavior?.conversionWindowStart ?? 0),
```